lab_2_tokenize_by_bpe/main.py

```diff
@@ -26,6 +35,15 @@ def collect_frequencies(
     :param end_of_word: a token that signifies the end of word
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if not isinstance(text, str) or not isinstance(start_of_word, str | None) or not isinstance(end_of_word, str):
+        return None
+    dictionary = {}
+    for i in text.split():
+        if prepare_word(i, start_of_word, end_of_word) is None:
+            return None
+        else:
```

Comment:
not need to use `else` if you return in if statement

Reply:
fixed